executor/insert_common.go

```diff
@@ -529,6 +559,165 @@ func (e *InsertValues) fillRow(ctx context.Context, row []types.Datum, hasValue
 	return row, nil
 }
 
+// isAutoNull can help judge whether a datum is AutoIncrement Null quickly.
+// This used to help lazyFillAutoIncrement to find consecutive N datum backwards for batch autoID alloc.
+func (e *InsertValues) isAutoNull(ctx context.Context, d types.Datum, col *table.Column) bool {
+	// autoID can find in RetryInfo.
+	if e.ctx.GetSessionVars().RetryInfo.Retrying {
```

Comment:
Redundant check? if `e.ctx.GetSessionVars().RetryInfo.Retrying` is true, it will be `continue` in line #633. https://github.com/pingcap/tidb/pull/11876/files#diff-0f31e77a0899889b749ff90c69acaaf6R633

Reply:
`for i+1 < length && e.isAutoNull(ctx, rows[i+1][colIdx], col)` 这里是向后寻找的小循环（寻找后继连续的 null ），新的 datum 在大循环中还没走